provider/agent.go

```diff
@@ -259,6 +259,66 @@ func agentResource() *schema.Resource {
 				ForceNew:    true,
 				Optional:    true,
 			},
+			"resources_monitoring": {
+				Type:        schema.TypeSet,
+				Description: "The resources monitoring configuration for this agent.",
+				ForceNew:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"memory": {
+							Type:        schema.TypeSet,
+							Description: "The memory monitoring configuration for this agent.",
+							ForceNew:    true,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enabled": {
+										Type:        schema.TypeBool,
+										Description: "Enable memory monitoring for this agent.",
+										ForceNew:    true,
+										Required:    true,
+									},
+									"threshold": {
+										Type:        schema.TypeInt,
+										Description: "The memory usage threshold in percentage at which to trigger an alert. Value should be between 0 and 100.",
+										ForceNew:    true,
+										Required:    true,
+									},
+								},
+							},
+						},
+						"volume": {
+							Type:        schema.TypeSet,
+							Description: "The volumes monitoring configuration for this agent.",
+							ForceNew:    true,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"path": {
+										Type:        schema.TypeString,
+										Description: "The path of the volume to monitor.",
```

Comment:
Validation of the volume can only be done in the agent
How do we plan to surface problems with accessing a volume - which may either be due to a permissions issue or an incorrect path?

Reply:
Indeed , for now the way it his done is :
- We validate that the path is not empty & this is an absolute one on tf-provider
- If the path is not accessible we ignore it on the agent side - but log it.

We can discuss the agent logic on the agent PR , but I feel like we do not want to block the worspace to start for monitoring , but definitely open to discuss it.

Reply:
> I feel like we do not want to block the worspace to start for monitoring , but definitely open to discuss it.

Do you mean that if the resource monitoring definition is somehow wrong it should still start the agent but log an error/warning?

Reply:
Sorry maybe was not precise enough - yes, but only for the situation on the which terraform itself is correct, validation is done BUT the volume is not present in the workspace, or not accessible.

For example `/user/coder` is a valid absolute path, but maybe do not exist in a given workspace - in that situation I would do what describe above - start the workspace & agent, but display a warning.

Reply:
Sounds good 👍